crates/sui-types/src/crypto.rs

```diff
@@ -68,25 +72,52 @@ pub type NetworkKeyPair = Ed25519KeyPair;
 pub type NetworkPublicKey = Ed25519PublicKey;
 pub type NetworkPrivateKey = Ed25519PrivateKey;
 
-pub const PROOF_OF_POSSESSION_DOMAIN: &[u8] = b"kosk";
 pub const DERIVATION_PATH_COIN_TYPE: u32 = 784;
 pub const DERVIATION_PATH_PURPOSE_ED25519: u32 = 44;
 pub const DERVIATION_PATH_PURPOSE_SECP256K1: u32 = 54;
 pub const TBLS_RANDOMNESS_OBJECT_DOMAIN: &[u8; 10] = b"randomness";
 
-// Creates a proof that the keypair is possesed, as well as binds this proof to a specific SuiAddress.
-pub fn generate_proof_of_possession<K: KeypairTraits>(
-    keypair: &K,
+/// Creates a proof of that the authority account address is owned by the
+/// holder of authority protocol key, and also ensures that the authority
+/// protocol public key exists. A proof of possession is an authority
+/// signature committed over the intent message `intent || message` (See
+/// more at [struct IntentMessage] and [struct Intent]) where the message is
+/// constructed as `authority_pubkey_bytes || authority_account_address`.
+pub fn generate_proof_of_possession(
+    keypair: &AuthorityKeyPair,
     address: SuiAddress,
-) -> <K as KeypairTraits>::Sig {
-    let mut domain_with_pk: Vec<u8> = Vec::new();
-    domain_with_pk.extend_from_slice(PROOF_OF_POSSESSION_DOMAIN);
-    domain_with_pk.extend_from_slice(keypair.public().as_bytes());
-    domain_with_pk.extend_from_slice(address.as_ref());
-    // TODO (joyqvq): Use Signature::new_secure
-    keypair.sign(&domain_with_pk[..])
+) -> AuthoritySignature {
+    let mut msg: Vec<u8> = Vec::new();
+    msg.extend_from_slice(keypair.public().as_bytes());
+    msg.extend_from_slice(address.as_ref());
+    AuthoritySignature::new_secure(
+        &IntentMessage::new(
+            Intent::default().with_scope(IntentScope::ProofOfPossession),
+            msg,
+        ),
+        &0, // Default to use epoch 0
+        keypair,
+    )
 }
 
+/// Verify proof of possession against the expected intent message,
+/// consisting of the protocol pubkey and the authority account address.
+pub fn verify_proof_of_possession(
+    pop: &narwhal_crypto::Signature,
+    protocol_pubkey: &narwhal_crypto::PublicKey,
+    sui_address: SuiAddress,
+) -> Result<(), SuiError> {
+    let mut msg = protocol_pubkey.as_bytes().to_vec();
```

Comment:
to improve performance of bls pk serde, we don't validate them during from_bytes but validate them explicitly where needed. currently we do that in the bls verify native function, and since we don't go through it now, we should do it here as well. simply call protocol_pubkey.validate()? here